docs/contribute/contribute.adoc

```diff
@@ -4,126 +4,169 @@ This document is meant for new OpenSCAP contributors and it describes how to
+$ git checkout -b master remotes/origin/master
+# and now create a new branch from the master branch which will contain your fix
+$ git checkout -b fix_455
+----
+
 
 == Fix the issue
-NOTE: OpenSCAP is licensed under LGPLv2. Any fixes or improvements must be licensed
-under the same license to be included. Check the COPYING file in the repository
-for more details.
+NOTE: OpenSCAP is licensed under _LGPL v2.1_. Any fixes or improvements must be
+licensed under the same license to be included. Check the COPYING file in the
+repository for more details.
 
 Now you can work on the fix. Try to create small commits which can be easily
 reviewed and make self-explaining commit messages. The
 link:http://chris.beams.io/posts/git-commit/[How to Write a Git Commit
 Message] article could help you with that. Nobody will review a pull request
 which contains a four thousand new lines of code.
 
-NOTE: Since we're fixing an issue number 455 make sure that your commit message
-contain a
+NOTE: Since you're fixing the issue number 455 make sure that your commit
+message contain a
 link:https://help.github.com/articles/closing-issues-via-commit-messages/[keyword]
 which will close the issue automatically when your pull request is merged.
 
-
 Let's say that you've fixed the issue, made a few commits to your local fix_455
 branch and you think it's ready to be review by other contributors. Before you
 push your local changes to your remote forked repository it's necessary to check
-if your changes will be applicable and won't be in a conflict with some work that
-  other contributors could publish while you were working on the fix.
+if your changes will be applicable and won't be in a conflict with some work
+that other contributors could publish while you were working on the fix.
+
+
+== Optional: Write an automated test for your code
+There is a big chance that the code you've fixed or the code that you've added
+has no test coverage. We encourage you to write a new test or extend the
+existing one to cover the changes/additions to OpenSCAP that you have made.
+It is not mandatory, so reviewer will not require you to write a test, but keep
+in mind that providing a test might uncover some unexpected issues with the
+code. We also run these tests on every pull request so adding a test for your
+fix or new feature might prevent someone from breaking it in the future. If you
+decided to add a test please see our
+link:testing.adoc[guide for writing and running tests].
 
 
 == Rebase before pull request
-If some other contributor pushed some code to the maint-1.0 branch while you was
-working of the fix then it's necessary to pull those changes and rebase our fix
-on top of them. First we need to make sure that our local maint-1.0 branch is
-up-to date.
+If some other contributor pushed some code to the `maint-1.2` branch while you
+were working on the fix then it's necessary to fetch those changes and rebase
```

Comment:
it's not necessary, it might be necessary if there are conflicts but we don't require ff-only merges AFAIK

Reply:
True.